scipio/src/executor.rs

```diff
@@ -293,6 +293,50 @@ impl ExecutorQueues {
     }
 }
 
+/// LocalExecutor factory, which can be used in order to configure the properties of a new
+/// [`LocalExecutor`].
+///
+/// Methods can be chained on it in order to configure it.
+///
+/// The [`spawn`] method will take ownership of the builder and create an [`io::Result`] to the
+/// [`LocalExecutor`] handle with the given configuration.
+///
+/// The [`LocalExecutor::spawn_default`] free function uses a Builder with default configuration and
+/// unwraps its return value.
+///
+/// You may want to use [`spawn`] instead of [`LocalExecutor::spawn_default`], when you want to
+/// recover from a failure to launch a thread, indeed the free function will panic where the Builder
+/// method will return a [`io::Result`].
+///
+/// # Examples
+///
+/// ```
+/// use scipio::LocalExecutorBuilder;
+///
+/// let builder = LocalExecutorBuilder::new();
+/// let ex = builder.spawn().unwrap();
+/// ```
+#[derive(Debug)]
+pub struct LocalExecutorBuilder {}
+
+impl LocalExecutorBuilder {
+    /// Generates the base configuration for spawning a [`LocalExecutor`], from which configuration
+    /// methods can be chained.
+    pub fn new() -> LocalExecutorBuilder {
+        LocalExecutorBuilder {}
+    }
+
+    /// Spawns a new [`LocalExecutor`] by taking ownership of the Builder, and returns an io::Result
+    /// to its JoinHandle.
+    pub fn spawn(&self) -> io::Result<LocalExecutor> {
```

Comment:
I don't like the fact that this returns a `Result`. One of the biggest advantage of this builder route is that if you don't do anything fancy it can't possibly fail so you don't have to check for errors.

Your examples don't check for errors, but I am pretty sure they would generate warnings (maybe there is a way to error out on warnings)

Reply:
We have to, because `spawn` can fail. All configuration methods on the builder are lazy: they never fail. If you pass an invalid value for CPU binding, `spawn` will fail.

Reply:
Right. But read that together with my previous comments about this being called spawn.

What we really need is a `create` function that doesn't allow any builder parameters.

This is similar to what `thread` does (except that they are indeed spawning, so theirs is called `spawn`.
You can get started really quickly by just calling `Local::create_executor()`, which doesn't ever return any errors.

For more complex scenarios you go to through the builder and that can return errors

Reply:
Alright I see what you mean. There is such a function: `LocalExecutor::make_default() -> LocalExecutor`. This is not configurable, uses the current thread and cannot fail.

The function above sits inside the builder so at this point some of the configuration the user provided may be faulty. 
For context this method as since been renamed `make()`. Another function `spawn()` exists too. It behaves the same way except that it also create a new thread to host the executor. It returns `io::Result<JoinHandle<()>>` as you'd expect.